ngraph/core/reference/include/ngraph/runtime/reference/function.hpp

```diff
@@ -0,0 +1,34 @@
+//*****************************************************************************
+// Copyright 2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <cmath>
+#include <cstddef>
+#include <vector>
+#include "ngraph/function.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            HostTensorVector function(const std::shared_ptr<Function>& function,
```

Comment:
Why do we use another syntax?
If I remember right all reference functions have next signature:  `void name(HostTensorVector in, HostTensorVector out, ...)`

Reply:
I tried to implement close to this code
```
file: inference-engine/tests/ngraph_functions/src/utils/ngraph_helpers.cpp
std::vector<std::vector<std::uint8_t>> interpreterFunction(const std::shared_ptr<Function> &function, const std::vector<std::vector<std::uint8_t>> &inputs,
                                                           std::vector<ngraph::element::Type_t> convertType)
```
Is it critical?

Reply:
No it is not critical, but will be better if we use one approach for all functions))

Reply:
" for all functions" Could you please give some examples? 
Do you mean ctors? Or what?

Reply:
Sure, here majority implementations don't have return value:
https://github.com/openvinotoolkit/openvino/tree/master/ngraph/core/reference/include/ngraph/runtime/reference

Reply:
fixed